Machine-Teaching-Examples/moab_pomdp_esn.ink

```diff
@@ -0,0 +1,147 @@
+###
+
+# MSFT Bonsai 
+# Copyright 2021 Microsoft
+# This code is licensed under MIT license (see LICENSE for details)
+
+# Moab sample illustrating usage of memory, as a Machine Teaching strategy,
+# for problems with missing state information, i.e. partially observable.
+# Using a history of states and actions can help when a sensor cannot read
+# all the relevant information to describe the dynamics.
+
+# We recommend including state information if you can measure it (velocity),
+# but adding memory is helpful when features are hard to define or don't
+# capture everything that is relevant.
+
+# Exporting a brain with memory handles the history for you. There is a
+# recurrent "reservoir" model with an internal state of size 100, called
```

Comment:
I wouldn't include the implementation details about ESN, reservoirs, etc -- they can change without notice.

Reply:
you shouldn't think of it as "last 100 states". It's "a 100-dimensional embedding summarizing the state history", which is both more useful and harder to reason about :).